osal/osal_common.h

```diff
@@ -6,62 +6,483 @@
+
+/****************************************************************************
+ * @fn      osal_sendmsg
+ *
+ * @brief  used to transmit a message to another socket. 
+ *
+ * input parameters
+ *  @param[in] sockfd socket file descriptor 
+ *  @param[in] msghdr structure contains message meta data.
+ *  @param[in] flags argument is formed by ORing one or many socket options 
+ *
+ * output parameters
+ * @return On success return the number of bytes sent on error -1 is returned 
+ *****************************************************************************/
 osal_ssize_t osal_sendmsg(int sockfd, const struct msghdr msg, int flags);
+
+/****************************************************************************
+ * @fn      osal_bind
+ *
+ * @brief  bind a name to a socket. 
+ *
+ * input parameters
+ *  @param[in] osal_sockfd socket file descriptor 
+ *  @param[in] osal_addr address to be binded
+ *  @param[in] addrlen address length
+ *
+ * output parameters
+ * @return On success 0 is returned.  On error, -1 is returned 
+ *****************************************************************************/
 osal_basetype_t osal_bind(int osal_sockfd, osal_sockaddr *osal_addr, 
 	osal_socklen addrlen);
 
+/****************************************************************************
+ * @fn      osal_sendto
+ *
+ * @brief  used to transmit a message to another socket. 
+ *
+ * input parameters
+ *  @param[in] sockfd socket file descriptor 
+ *  @param[in] buf message buffer
+ *  @param[in] len data buffer len 
+ *  @param[in] flags argument is formed by ORing one or many socket options 
+ *  @param[in] dest_addr destination address 
+ *  @param[in] addrlen address length
+ *
+ * output parameters
+ * @return On success return the number of bytes sent on error -1 is returned 
+ *****************************************************************************/
 osal_ssize_t osal_sendto(osal_basetype_t sockfd, const void *buf, size_t len, int flags,
                       const osal_sockaddr *dest_addr, osal_socklen addrlen);
+
+/****************************************************************************
+ * @fn      osal_inet_pton
+ *
+ * @brief  This function converts the character string src into a network 
+ *         address structure in the af address family 
+ *
+ * input parameters
+ *  @param[in] af address family 
+ *  @param[in] src character string 
+ *  @param[out] dst network address structure
+ *
+ * output parameters
+ * @return 1 on success
+ *         0 if src does not contain a character string representing a valid
+ *           network address in the specified address family
+ *        -1 if af does not contain a valid address family
+ *****************************************************************************/
 osal_basetype_t osal_inet_pton(int af, const char *src, void *dst);
+
+/****************************************************************************
+ * @fn      osal_select
```

Comment:
The select API is not used in select contexts, since works with only one file descriptor in coapclient.c and coapserver.c in the library sources. It can be entirely replaced with blocking recvfrom(), because there isn't any other "readable" descriptor in the implementation.